### PR TITLE
feat: switch workers to notifications v2

### PR DESCRIPTION
**Pull request by idoshamun**
Delete old workers and use the new ones.
I validated that switching to existing topic to v2 will not have any significant impact (potentially some notifications may not be delivered during deployment but that's fine given the alternative).

I also added metrics 😎

Sidenote: switched to `toHaveBeenCalledTimes` as the other function is deprecated

WT-2045 #done

### CODE REVIEW

File: src/common/metrics.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { opentelemetry } from '../telemetry/opentelemetry';
+
+export function getNotificationFailedCounter(): opentelemetry.Counter<{
+  channel: string;
+}> {
+  const meter = opentelemetry.metrics.getMeter('api-bg');
```
Comment (rebelchris) [inline src/common/metrics.ts:6]: You love to see it

File: src/common/metrics.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { opentelemetry } from '../telemetry/opentelemetry';
+
+export function getNotificationFailedCounter(): opentelemetry.Counter<{
+  channel: string;
```
Comment (rebelchris) [inline src/common/metrics.ts:4]: Don't you want to use the channel somewhere in the counter?

File: src/workers/newNotificationV2Mail.ts [hunk 3]
```diff
@@ -695,6 +697,7 @@ const worker: Worker = {
         },
         'failed to send notification email',
       );
+      counter.add(1, { channel: 'email' });
```
Comment (rebelchris) [inline src/workers/newNotificationV2Mail.ts:700]: Oh I see you export the counter :) 
That makes more sense now.

**Review by rebelchris — APPROVED** [review]
Nothing blocking from my side.

File: src/workers/newNotificationV2Mail.ts (hunk 3 above)
Comment (idoshamun) [reply src/workers/newNotificationV2Mail.ts:700]: Type safety rules!!

**Pull request merged**